File: Login System ALL VERSIONS/Login System.py
def register(users):
    username = input("Enter a username: ")
    if username in users:
        print("Username already exists.")
        return
    password = input("Enter a password: ")
    users[username] = password
    print("Registration successful.")

def login(users):
    username = input("Enter your username: ")
    password = input("Enter your password: ")
    if username in users and users[username] == password:
        print("Login successful.")
    else:
        print("Invalid username or password.")

def main():
    users = {}
    while True:
        print("\n1. Register")
        print("2. Login")
        print("3. Exit")
        choice = input("Enter your choice: ")

        if choice == "1":
            register(users)
        elif choice == "2":
            login(users)
        elif choice == "3":
            print("Exiting...")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()

File: Login System ALL VERSIONS/LoginSystem.java
import java.io.*;
import java.util.*;

public class LoginSystem {
    private static final String USERS_FILE = "users.txt";

    public static void main(String[] args) {
        Map<String, String> users = loadUsers();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\n1. Register");
            System.out.println("2. Login");
            System.out.println("3. Exit");
            System.out.print("Enter your choice: ");
            String choice = scanner.nextLine();

            switch (choice) {
                case "1":
                    register(users, scanner);
                    break;
                case "2":
                    login(users, scanner);
                    break;
                case "3":
                    System.out.println("Exiting...");
                    System.exit(0);
                default:
                    System.out.println("Invalid choice. Please try again.");
            }
        }
    }

    private static Map<String, String> loadUsers() {
        Map<String, String> users = new HashMap<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(USERS_FILE))) {
            String line;
            while ((line = reader.readLine()) != null) {
                String[] parts = line.split(" ");
                users.put(parts[0], parts[1]);
            }
        } catch (IOException e) {
            // File does not exist yet, return empty map
        }
        return users;
    }

    private static void saveUser(String username, String password) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(USERS_FILE, true))) {
            writer.println(username + " " + password);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static void register(Map<String, String> users, Scanner scanner) {
        System.out.print("Enter a username: ");
        String username = scanner.nextLine();
        if (users.containsKey(username)) {
            System.out.println("Username already exists.");
            return;
        }
        System.out.print("Enter a password: ");
        String password = scanner.nextLine();
        users.put(username, password);
        saveUser(username, password);
        System.out.println("Registration successful.");
    }

    private static void login(Map<String, String> users, Scanner scanner) {
        System.out.print("Enter your username: ");
        String username = scanner.nextLine();
        System.out.print("Enter your password: ");
        String password = scanner.nextLine();
        if (users.containsKey(username) && users.get(username).equals(password)) {
            System.out.println("Login successful.");
        } else {
            System.out.println("Invalid username or password.");
        }
    }
}
